backend/canisters/user_index/impl/src/lib.rs

```diff
@@ -182,6 +190,8 @@ struct Data {
     pub internet_identity_canister_id: CanisterId,
     #[serde(default)]
     pub next_user_upgrade_started: bool,
+    #[serde(default)]
+    pub referrer_codes: HashMap<String, ReferralType>,
```

Comment:
I think we should store more data about these codes such as when they were created and when the were used.